adafruit_ble/__init__.py

```diff
@@ -181,6 +183,7 @@ def start_advertising(self, advertisement, scan_response=None, interval=0.1, tim
             scan_response.tx_power = self.tx_power
         if scan_response:
             scan_response_bytes = bytes(scan_response)
+        timeout = 0 if timeout is None else timeout
```

Comment:
You can remove this from here and put it inside the call to `self._adapter.start_advertising(). Sorry I didn't make that clearer.
```python
        self._adapter.start_advertising(
            advertisement_bytes,
            scan_response=scan_response_bytes,
            connectable=advertisement.connectable,
            interval=interval,
            timeout=0 if timeout is None else timeout,
        )
```